src/helpers/loadDocument.js

```diff
@@ -29,7 +29,12 @@ export default (state, dispatch, extraOptions) => {
             partRetriever.setErrorCallback(fireError);
           }
 
-          dispatch(actions.openElement('progressModal'));
+          // Delaying progress modal for 2 sec, so that UI will wait and
+          // fade-in the progress modal if the file downloading is too slow
+          setTimeout(() => {
+            dispatch(actions.openElement('progressModal'));
+          }, 2000);
```

Comment:
Can we just use css transition or animation delay of 2s instead of using setTimeout?
I feel like a pure css solution is better. Something like
```
  &.open {
    opacity: 1;
    transition: opacity 2s 1s;
  }

  &.closed {
    opacity: 0;
  }
```
The transition of opacity will wait for 2 secs before starting.
If we do this we also don't need the extra logic in openElements, because if the document is loaded or if there's an error loading a document, the progress modal will be closed immediately.

Reply:
Yeah, sure we can do that,  I'll revert add CSS